File: Data Structures & Algorithms/maximum-subarray/submission-3.java
// Given an integer array nums, find the subarray with the largest sum, and return its sum.

class Solution {
    public int maxSubArray(int[] nums) {
        // Initialize max sum and curr sum to keep track of the maxValues
        int maxSum = Integer.MIN_VALUE;
        int currSum = 0;

        // Iterate through the array
        for(int num: nums){
            // If currSum is less than 0, we reset. Reason? We want don't want to add -ve values as it hurts our maxValue
            if(currSum < 0){
                currSum = 0;
            }

            // Add num to currSum
            currSum += num;

            // Keep the max of currSum and maxSum
            maxSum = Math.max(currSum, maxSum);
        }

        return maxSum;
    }
}

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-3.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        result = [0] * len(arr)
        maxValue = -1
        for i in range(len(arr)-1, -1,-1):
            result[i] = maxValue
            maxValue = max(maxValue, arr[i])
        return result
            